### PR TITLE
add WFS 3 /api endpoint to docs, and reformat some tables

### PR DESCRIPTION
**Pull request by philvarner**

### CODE REVIEW

File: api-spec/api-spec.md [hunk 1]
```diff
@@ -22,11 +22,12 @@ The WFS 3 and STAC APIs follow a RESTful model.  A core principal of this is the
...
 
-| Endpoint      | Returns          | Description        |
-| ------------ | ------------- | ---------------------- |
-| / | JSON        | Landing page, links to API capabilities |
-| /conformance | JSON | Info about standards the API conforms to       |
-| /collections | Collections | List of Collections contained in the catalog |
+| Endpoint     | Returns       | Description |
+| ------------ | ------------- | ----------- |
+| /            | JSON          | Landing page, links to API capabilities |
+| /api         | JSON          | API definition |
```
Comment (philvarner) [inline api-spec/api-spec.md]: this endpoint was missing from this doc, but in the WFS 3 spec

**Comment by philvarner** [comment]
I'm looking at https://rawcdn.githack.com/opengeospatial/WFS_FES/3.0.0-draft.1/docs/17-069.html#_api_definition_2 
`The server SHALL support the HTTP GET operation at the path /api.`

conformance was already there, i just cleaned up the formatting

**Comment by joshfix** [comment]
> There's no /api in https://stacspec.org/STAC-api.html and it seems it's not required in WFS, so we may leave it out. Let's wait a little with this until we know more about OGC API Commons. I'll take care to update with #472
> 
> Adding conformance though is fine, so maybe remove /api and we'll update to include /api once I backport WFS and if it's included there.

I went ahead and approved as it seems to be legit.  Back to you @m-mohr

**Comment by m-mohr** [comment]
I just discussed this with Clemens, one of the authors of the WFS spec. 
/api is indeed required by the WFS spec, but is not (and probably won't be in the future) included in the OpenAPI spec of WFS. So, I'm feeling that it is confusing if /api is explained here, but not mentioned anywhere ein the OpenAPI documentation.

That said, now I'd even go further and add explanation here that /api is required by the WFS specification (with link to the requirement) and then explain here that it is not part of the OpenAPI spec, but still needs to be implemented.

**Comment by joshfix** [comment]
That seems super confusing.  So WFS is publishing an OpenAPI spec for WFS, but not including required endpoints?  Do we really need to explain WFS documentation in our STAC documentation, or can we just require the endpoint?

**Comment by m-mohr** [comment]
Yes, the reason was that /api returns the API document itself, which you are looking at anyway. I made the point that requiring an endpoint that is not in the OpenAPI file is confusing, but we didn't get to a consensus. Maybe open an issue in the WFS issue tracker so that they know I'm not the only one that finds this super confusing.

**Comment by philvarner** [comment]
Added language around /api missing.  will file issue with WFS 3.

**Comment by philvarner** [comment]
This is the issue that caused it to be removed https://github.com/opengeospatial/WFS_FES/issues/7

**Comment by philvarner** [comment]
Filed https://github.com/opengeospatial/WFS_FES/issues/236

It as a requirement to implement the /api endpoint according to the table in the Abstract section of https://rawcdn.githack.com/opengeospatial/WFS_FES/3.0.0-draft.1/docs/17-069.html and 7.3. API definition. However, this is endpoint is not present in the openapi.yaml documentation. I see that it was removed as a result of #7 , however, as an implementer, I find this confusing. Just adding a definition of the endpoint and that it returns an entity of type application/vnd.oai.openapi would make this clear.

**Comment by m-mohr** [comment]
Thanks for finding that issue and adding a short sentence about the absence of /api in the OpenAPI spec. Approved and merged.

**Pull request merged**